sql/exporter.go

```diff
@@ -4,27 +4,81 @@ import (
 	"fmt"
 	"io"
 	"strings"
+	"sort"
 )
 
 const (
-	hierarchySql = "insert into hierarchy (hierarchy_id, hierarchy_name) values ('%s', %s);\n"
-	entrySql     = "insert into hierarchy_entry (hierarchy_id, entry_code, parent_code, code_name, abbreviation, description, level) values ('%s', '%s', %s, %s, %s, %s, %d);\n"
+	hierarchySql = "insert into hierarchy (hierarchy_id, hierarchy_name) values (%s, %s);\n"
+	areaSql = "insert into hierarchy_area_type (id, name, level) select %s, %s, %d where not exists (select id from hierarchy_area_type where id=%[1]s);\n"
+	entrySql     = "insert into hierarchy_entry (hierarchy_id, entry_code, parent_code, name, area_type) values (%s, %s, %s, %s, %s);\n"
 )
 
+
 // write sql to the given writer.
 // Please note that there is no error handling of failed writes
 // This is a stand-alone command line tool, so errors can just be reported to the user
-func WriteInserts(writer io.Writer, hierarchy *Hierarchy) {
+func WriteSql(writer io.Writer, hierarchy *Hierarchy) {
 
 	if len(hierarchy.Id) == 0 {
 		panic("Cannot write sql for a hierarchy without an id!")
 	}
-	io.WriteString(writer, fmt.Sprintf(hierarchySql, hierarchy.Id, quote(hierarchy.Names["en"])))
+	io.WriteString(writer, fmt.Sprintf(hierarchySql, quote(hierarchy.Id), quote(hierarchy.Names["en"])))
```

Comment:
This is writing sql statements to a file, so we can't avoid quotes. There is no actual interaction with a db.